Quotes added around activity function string parameter in isolated mode

### Description

Using Microsoft.Azure.Functions.Worker.Extensions.DurableTask version 1.1.0, when you pass a string parameter to an activity function (as a parameter with the [ActivityTrigger] attribute), the activity function receives it with quotes added around it.

I have an app that uses Durable Functions isolated mode in .NET 7, and  when I upgraded the Microsoft.Azure.Functions.Worker.Extensions.DurableTask package to 1.1.0 this started happening. Thinking it might be something to do with my code, I used the new project wizard in Visual Studio to create a new Durable Functions isolated project and chose to have it create an orchestration function, so I now had the example "SayHello" app. Running this, it generates the expected `Saying hello to Tokyo`, `Saying hello to Seattle`, `Saying hello to London` messages. But when I updated the NuGet packages so that I'm now using Microsoft.Azure.Functions.Worker.Extensions.DurableTask 1.1.0, it now generates `Saying hello to "Tokyo"`, `Saying hello to "Seattle"`, `Saying hello to "London"`. Also, if you break in the SayHello function, you'll see that the variable "name" doesn't just contain, say, "Tokyo," it contains "\\"Tokyo\\"".

> A clear and concise description of what you expected to happen.

I expected that the activity function would receive the string passed in the CallAcivityAsync call unchanged, as it did in previous versions of the library.

### Actual behavior

The activity function received the string with quotes added around it.

### Relevant source code snippets

You can get this code by just creating the default app with the wizard in Visual Studio, but here are the relevant parts of the code. The orchestration function calls the activity function here:

```
outputs.Add(await context.CallActivityAsync<string>(nameof(SayHello), "Tokyo"));
```

And then the activity function starts with:

```
public static string SayHello([ActivityTrigger] string name, FunctionContext executionContext)
{   
    ...
```

In that function, "name" contains (for the above call) `"Tokyo"`.

### Known workarounds

None that I know of, except for stripping the quotes out of any strings passed to activity functions.

### App Details

Microsoft.Azure.Functions.Worker.Extensions.DurableTask 1.1.0
Azure Functions runtime version 4
Programming language used: C#

I have the same problem. However, if I downgrade `Microsoft.Azure.Functions.Worker.Extensions.DurableTask` to `1.0.4` or if I upgrade to `1.2.0-rc.1` then my project won't build because the `Microsoft.DurableTask.Entities` namespace no longer exists.
```
error CS0234: The type or namespace name 'Entities' does not exist in the namespace 'Microsoft.DurableTask' (are you missing an assembly reference?)
error CS0246: The type or namespace name 'TaskEntityDispatcher' could not be found (are you missing a using directive or an assembly reference?)
error CS0246: The type or namespace name 'EntityTriggerAttribute' could not be found (are you missing a using directive or an assembly reference?)
error CS0246: The type or namespace name 'EntityTrigger' could not be found (are you missing a using directive or an assembly reference?)
```

--- a/src/Worker.Extensions.DurableTask/ActivityInputConverter.cs
+++ b/src/Worker.Extensions.DurableTask/ActivityInputConverter.cs
@@ -25,6 +25,11 @@ internal class ActivityInputConverter : IInputConverter
             throw new ArgumentNullException(nameof(context));
         }
 
+        if (context.Source is null)
+        {
+            return new(ConversionResult.Success(null));
+        }
+
         if (context.Source is not string source)
         {
             throw new InvalidOperationException($"Expected converter source to be a string, received {context.Source?.GetType()}.");

--- a/src/Worker.Extensions.DurableTask/ActivityInputConverter.cs
+++ b/src/Worker.Extensions.DurableTask/ActivityInputConverter.cs
@@ -1,0 +1,36 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.Azure.Functions.Worker.Converters;
+using Microsoft.DurableTask.Worker;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.Azure.Functions.Worker.Extensions.DurableTask;
+
+internal class ActivityInputConverter : IInputConverter
+{
+    private readonly DurableTaskWorkerOptions options;
+
+    public ActivityInputConverter(IOptions<DurableTaskWorkerOptions> options)
+    {
+        this.options = options?.Value ?? throw new ArgumentNullException(nameof(options));
+    }
+
+    public ValueTask<ConversionResult> ConvertAsync(ConverterContext context)
+    {
+        if (context is null)
+        {
+            throw new ArgumentNullException(nameof(context));
+        }
+
+        if (context.Source is not string source)
+        {
+            throw new InvalidOperationException($"Expected converter source to be a string, received {context.Source?.GetType()}.");
+        }
+
+        object? value = this.options.DataConverter.Deserialize(source, context.TargetType);
+        return new(ConversionResult.Success(value));
+    }
+}

--- a/src/Worker.Extensions.DurableTask/ActivityTriggerAttribute.cs
+++ b/src/Worker.Extensions.DurableTask/ActivityTriggerAttribute.cs
@@ -3,7 +3,9 @@
 
 using System;
 using System.Diagnostics;
+using Microsoft.Azure.Functions.Worker.Converters;
 using Microsoft.Azure.Functions.Worker.Extensions.Abstractions;
+using Microsoft.Azure.Functions.Worker.Extensions.DurableTask;
 
 namespace Microsoft.Azure.Functions.Worker;
 
@@ -12,6 +14,8 @@ namespace Microsoft.Azure.Functions.Worker;
 /// </summary>
 [AttributeUsage(AttributeTargets.Parameter)]
 [DebuggerDisplay("{Activity}")]
+[InputConverter(typeof(ActivityInputConverter))]
+[ConverterFallbackBehavior(ConverterFallbackBehavior.Disallow)]
 public sealed class ActivityTriggerAttribute : TriggerBindingAttribute
 {
     /// <summary>

--- a/src/Worker.Extensions.DurableTask/DurableClientAttribute.cs
+++ b/src/Worker.Extensions.DurableTask/DurableClientAttribute.cs
@@ -1,13 +1,17 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
+using Microsoft.Azure.Functions.Worker.Converters;
 using Microsoft.Azure.Functions.Worker.Extensions.Abstractions;
+using Microsoft.Azure.Functions.Worker.Extensions.DurableTask;
 
 namespace Microsoft.Azure.Functions.Worker;
 
 /// <summary>
 /// Azure Functions attribute for binding a function parameter to a Durable Task client object.
 /// </summary>
+[InputConverter(typeof(DurableTaskClientConverter))]
+[ConverterFallbackBehavior(ConverterFallbackBehavior.Disallow)]
 public sealed class DurableClientAttribute : InputBindingAttribute
 {
     /// <summary>

--- a/src/Worker.Extensions.DurableTask/DurableTaskExtensionStartup.cs
+++ b/src/Worker.Extensions.DurableTask/DurableTaskExtensionStartup.cs
@@ -58,7 +58,6 @@ public sealed class DurableTaskExtensionStartup : WorkerExtensionStartup
 
         applicationBuilder.Services.Configure<WorkerOptions>(o =>
         {
-            o.InputConverters.RegisterAt<DurableTaskClientConverter>(0);
             o.InputConverters.Register<OrchestrationInputConverter>();
         });
 

--- a/src/Worker.Extensions.DurableTask/HTTP/HttpHeadersConverter.cs
+++ b/src/Worker.Extensions.DurableTask/HTTP/HttpHeadersConverter.cs
@@ -28,7 +28,7 @@ internal class HttpHeadersConverter : JsonConverter<IDictionary<string, StringVa
         var valueList = new List<string>();
         while (reader.Read() && reader.TokenType != JsonTokenType.EndObject)
         {
-            string? propertyName = reader.GetString();
+            string propertyName = reader.GetString()!;
 
             reader.Read();
 
@@ -42,7 +42,7 @@ internal class HttpHeadersConverter : JsonConverter<IDictionary<string, StringVa
             {
                 while (reader.Read() && reader.TokenType != JsonTokenType.EndArray)
                 {
-                    valueList.Add(reader.GetString());
+                    valueList.Add(reader.GetString()!);
                 }
 
                 values = new StringValues(valueList.ToArray());
@@ -62,8 +62,7 @@ internal class HttpHeadersConverter : JsonConverter<IDictionary<string, StringVa
     {
         writer.WriteStartObject();
 
-        var headers = (IDictionary<string, StringValues>)value;
-        foreach (var pair in headers)
+        foreach (KeyValuePair<string, StringValues> pair in value)
         {
             if (pair.Value.Count == 1)
             {

--- a/src/Worker.Extensions.DurableTask/Worker.Extensions.DurableTask.csproj
+++ b/src/Worker.Extensions.DurableTask/Worker.Extensions.DurableTask.csproj
@@ -37,7 +37,7 @@
   </PropertyGroup>
   
   <ItemGroup>
-    <PackageReference Include="Microsoft.Azure.Functions.Worker.Core" Version="1.15.0" />
+    <PackageReference Include="Microsoft.Azure.Functions.Worker.Core" Version="1.16.0" />
     <PackageReference Include="Microsoft.Azure.Functions.Worker.Extensions.Abstractions" Version="1.3.0" />
     <PackageReference Include="Microsoft.DurableTask.Client.Grpc" Version="1.1.0" />
     <PackageReference Include="Microsoft.DurableTask.Worker.Grpc" Version="1.1.0" />

